src/billing-reports/consommation-report.component.tsx

```diff
@@ -0,0 +1,289 @@
+import { exportSingleRecordToPDF, exportToExcel, formatValue } from './utils/download-utils';
+import styles from './billing-reports.scss';
+import { useTranslation } from 'react-i18next';
+import { formatToYMD } from './utils/download-utils';
+
+interface ReportRecord {
+  column: string;
+  value: string | string[];
+}
+
+interface ReportRow {
+  record: ReportRecord[];
+}
+
+interface Filters {
+  startDate: string;
+  endDate: string;
+  company: string;
+}
+
+const ConsommationReport: React.FC = () => {
+  const { t } = useTranslation();
+
+  const [results, setResults] = useState<ReportRow[]>([]);
+  const [columns, setColumns] = useState<string[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(50);
+  const [totalRecords, setTotalRecords] = useState(0);
+  const [currentFilters, setCurrentFilters] = useState<Filters | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const recordMap = new Map<string, ReportRecord[]>();
+
+  const getValue = (record: ReportRecord[], column: string): string => {
+    const found = record.find((item) => item.column === column);
+    const value = found?.value;
+    return formatValue(value);
+  };
+
+  const hiddenColumns = [
+    'global_amount',
+    'patientdue',
+    'insurancedue',
+    'paid_amount',
+    'bill_status',
+    'admission_type',
+    'global_bill_status',
+    'collectorname',
+  ];
+
+  const handleSearch = async (filters: Filters, pageNum = 1, pageSize = 50) => {
+    setLoading(true);
+    setErrorMessage(null);
+    try {
+      const formattedStart = formatToYMD(filters.startDate);
+      const formattedEnd = formatToYMD(filters.endDate);
+
+      const { results, total } = await fetchConsommationReport(
+        formattedStart,
+        formattedEnd,
+        filters.company,
+        pageNum,
+        pageSize,
+      );
+
+      if (results.length > 0) {
+        const columnNames = results[0].record.map((item) => item.column);
+        setColumns(columnNames);
+      }
+
+      setResults(results);
+      setTotalRecords(total);
+      setPage(pageNum);
+      setPageSize(pageSize);
+      setCurrentFilters(filters);
+    } catch (error) {
+      console.error('Error fetching report:', error);
+      setErrorMessage(t('errorFetchingReport', 'Failed to load report data.'));
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleExportClick = async () => {
+    if (!currentFilters) return;
+
+    setLoading(true);
+    try {
+      const { startDate, endDate, company } = currentFilters;
+      const allResults = await fetchAllConsommationReport(formatToYMD(startDate), formatToYMD(endDate), company);
+
+      exportToExcel(columns, allResults, getValue, 'Consommations-report.xlsx');
+    } catch (error) {
+      console.error('Export failed:', error);
+      setErrorMessage(t('errorExportingExcel', 'Failed to export to Excel.'));
```

Comment:
@lucyjemutai I would advise we provide the user with a useful error through the OpenMRS snackbar context, change other occurrences of this as well

```suggestion
catch (error) {
  showSnackbar({
    message: t('errorFetchingReport', 'Failed to load report data.'),
    type: 'error'
  });
}
```